### ISSUE TITLE
[semi] does not detect missing semicolons in interface definitions or in/around type aliases

### ISSUE THREAD

**Issue by lucaswerkmeister** [typescript-eslint/typescript-eslint#714]
**Repro**

```JSON
{
  "rules": {
    "semi": ["error", "always"]
  }
}
```

```TS
interface I {
    f(): void // <-- no comma
}

type foo = {
    bar: string // <-- no comma
    baz: number // <-- no comma
} // <-- no comma
```

**Expected Result**

Errors on all the marked lines. (I wasn’t sure whether to expect a semicolon after the type alias at first, but it looks like an assignment and the [TypeScript handbook](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-aliases) puts semicolons there.)

**Actual Result**

No errors.

**Additional Info**

This was already brought up by @Bkucera in https://github.com/typescript-eslint/typescript-eslint/pull/461#issuecomment-501346452; @bradzacher suggested raising a concrete issue, which as far as I can tell wasn’t done so far, so here it is.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.1-alpha.5` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.0.1` |
| `node`                             | `12.6.0` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
This is expected.

Please use `@typescript-eslint/member-delimiter-style` to decide what delimiters should be used within type/interfaces.
Please use `@typescript-eslint/semi` to support checking for semis at the end of typescript specific statements.

**Comment by lucaswerkmeister** [comment]
What about the missing semicolon after the type alias? I assume that’s not covered by member-delimiter-style. (But thanks for the pointer!)

**Comment by lucaswerkmeister** [comment]
I’ve also opened #715, perhaps that helps to prevent someone else from making the same mistake as me :)
